app/src/main/java/com/example/mvvmcleanarchitecture/features/episodes/all/presentation/EpisodeFragment.kt

```diff
@@ -11,55 +9,22 @@ import org.koin.android.ext.android.inject
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class EpisodeFragment :
-    BaseFragment<EpisodeViewModel, FragmentEpisodeBinding>(R.layout.fragment_episode),
+    BaseFragment<EpisodeViewModel, FragmentEpisodeBinding>(
```

Comment:
Mam 1 pytanie odnośnie klasy bazowej dla fragmentu:
w BaseFragment mamy metodę 
`    private fun bindViewModelToLifecycle(){
        lifecycle.addObserver(viewModel)
    }`

Problem polega że IDE zwraca mi to jako błąd, żę viewModel jest typu T a addObserver wymaga podania LifecycleObserver. Czy realnie ta metoda jest potrzebna do poprawnego działania aplikacji ?

Reply:
@holakmateusz zobacz czy klasa `BaseViewModel` implementuje interfejs `DefaultLifecycleObserver`. Bardzo możliwe, że omyłkowo gdzieś Ci to uciekło i dlatego masz ten problem. Sama metoda `bindViewModelToLifecycle` sprawia, że po stronie view modelu mamy dostęp do cyklu życia aktywności fragmentu, który z tego view modelu korzysta. W samej apce chyba nie jest to nigdzie realnie używane, natomiast podałem to jako taka ciekawostka jak można właśnie takie zmiany lifecycle obserwować po stronie view modelu.